### PR TITLE
Compliance frameworks, groups, framework items and library items.

### PR DESCRIPTION
**Pull request by asaba-hashi**
This PR implements the basic resources for:

* Frameworks
* Groups (Sections in the web UI)
* Framework Items (Requirements in the web UI)
* Library Items (Controls in the web UI)

This PR does not implement:

- Linking Questions to Framework or Library Items
- Linking Controls to Library Items

Links will likely need their own association style resources, so I'm going to implement linking in the next PR.

### CODE REVIEW

**Comment by zemberdotnet** [comment]
@asaba-hashi Thanks again for the great contributions! I'll begin working on the review today.

File: jupiterone/modifiers.go [hunk 1]
```diff
@@ -180,6 +181,65 @@ func (j jsonIgnoreDiff) PlanModifyList(ctx context.Context, req planmodifier.Lis
...
+	for k, v := range req.PlanValue.Elements() {
+		o, ok := oldValues[k]
+		if !ok {
+			// something new, go ahead with the plan
+			return
+		}
+
+		s, ok := v.(types.String)
+		if !ok {
+			// this is very bad and shouldn't haven't gotten past validation
```
Comment (zemberdotnet) [inline jupiterone/modifiers.go:204]: Is there any kind of error we should return or publish?

File: jupiterone/resource_framework.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"web_link": schema.StringAttribute{
+				Optional:    true,
+				Description: "A URL for referencing additional information about the framework",
+				// TODO: basic URL pattern matching
+				//Validators: []validator.String{
+				//},
```
Comment (zemberdotnet) [inline jupiterone/resource_framework.go:130]: I'll make an issue for this.

File: jupiterone/resource_framework.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+// ModifyPlan implements resource.ResourceWithModifyPlan
+func (*ComplianceFrameworkResource) ModifyPlan(ctx context.Context, _ resource.ModifyPlanRequest, _ *resource.ModifyPlanResponse) {
+	tflog.Info(ctx, "TODO")
+}
+
+// ConfigValidators implements resource.ResourceWithConfigValidators
+func (*ComplianceFrameworkResource) ConfigValidators(ctx context.Context) []resource.ConfigValidator {
+	tflog.Info(ctx, "TODO")
+	return nil
+}
```
Comment (zemberdotnet) [inline jupiterone/resource_framework.go:156]: Should I make issues for these as well?

**Review by zemberdotnet — DISMISSED** [review]
@asaba-hashi Just a few small notes. Everything looks good and I can merge if none of these seem like a concern.

File: jupiterone/modifiers.go (hunk 1 above)
Comment (asaba-hashi) [reply jupiterone/modifiers.go:204]: > Is there any kind of error we should return or publish?

I have added a warning for this to the diagnostics.

File: jupiterone/resource_framework.go (hunk 3 above)
Comment (asaba-hashi) [reply jupiterone/resource_framework.go]: I just removed these. I added them thinking that `version` was bumped by the server on changes like `version` on rules, but it turns out it's just a user configured field on these.

File: jupiterone/resource_framework.go (hunk 2 above)
Comment (asaba-hashi) [reply jupiterone/resource_framework.go:128]: :+1: to file an issue. The API will accept a lot more than the web UI, so it's unclear to me how much the terraform provider itself should be enforcing.

File: docs/resources/framework.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+### Required
+
+- `framework_type` (String) Whether this is a standard, benchmark, or questionnaire
+- `name` (String) The framework's name
+- `version` (String) Version of the framework itself (not an J1 API version)
+
+### Optional
+
+- `scope_filters` (List of String) JSON encoded filters for scoping the framework.
```
[the remark this answers is not in the document]
Comment (asaba-hashi) [reply docs/resources/framework.md:61]: Added a `scope_filter` from the the docs to the example resources.

**Review by zemberdotnet — APPROVED** [review]
@asaba-hashi Ready to merge if you are.

**Comment by asaba-hashi** [comment]
> @asaba-hashi Ready to merge if you are.

This is RTM 👍 .

**Pull request merged**